x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java

```diff
@@ -125,11 +127,11 @@ private void createNativeProcess(String jobId, AnalyticsProcessConfig analyticsP
             analyticsBuilder.build();
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
-            LOGGER.warn("[{}] Interrupted while launching data frame analytics process", jobId);
+            logger.warn("[{}] Interrupted while launching data frame analytics process", jobId);
         } catch (IOException e) {
             String msg = "[" + jobId + "] Failed to launch data frame analytics process";
-            LOGGER.error(msg);
-            throw ExceptionsHelper.serverError(msg, e);
+            logger.error(msg);
+            throw ExceptionsHelper.serverError(msg + " on [" + clusterService.getNodeName() + "]", e);
```

Comment:
```suggestion
            throw ExceptionsHelper.serverError("{} on [{}]", e, msg, clusterService.getNodeName());
```
or simply `nodeName` if previous suggestions are accepted.